### ISSUE TITLE
Native window/desktop switching moves windows between spaces

### ISSUE THREAD

**Issue by lalithsuresh** [koekeishiya/yabai#741]
To reproduce this:

1. Create multiple spaces.
2. Keep an app like a browser in space 1
3. Have another an app in space 8 (say, a text editor)
4. Cmd+Tab to pick the browser
5. This should trigger OSX's native desktop change animation to bring space 1 (and desktop 1) into focus
6. The app has moved from space 8 to space 2. Other windows have moved spaces too (if I were to guess, they all moved one space to the right)

For step 3, I can avoid using cmd+tab, but this behavior is unavoidable when for example, clicking openable links that triggers the native desktop-switch animation to bring the browser into focus.

Basically, it seems like the native desktop change animation is garbling the assignment of windows to spaces. Any way to fix this? I'd like the windows to be pinned to specific spaces unless I reassign them myself.

I'm on yabai-v3.1.0 and on MacOS Catalina 10.15.7.

**Comment by lalithsuresh** [comment]
Also reproducible on yabai-v3.3.4

**Comment by koekeishiya** [comment]
Check this setting:

In the Mission Control preferences pane in System Preferences, the setting "Automatically rearrange Spaces based on most recent use" should be disabled for commands that rely on the ordering of spaces to work reliably.

**Comment by lalithsuresh** [comment]
That fixes it. Thanks!
